src/request.rs

```diff
@@ -246,6 +277,112 @@ async fn write_header<C: Write>(c: &mut C, key: &str, value: &str) -> Result<(),
     Ok(())
 }
 
+/// The request body
+pub trait RequestBody {
+    /// Get the length of the body if known
+    ///
+    /// If the length is known, then it will be written in the `Content-Length` header,
+    /// chunked encoding will be used otherwise.
+    fn len(&self) -> Option<usize> {
+        None
+    }
+
+    /// Write the body to the provided writer
+    async fn write<W: Write>(&self, writer: &mut W) -> Result<(), W::Error>;
+}
+
+impl RequestBody for () {
```

Comment:
I'm wondering if this could be impl<T> RequestBody for Option<T> where T: RequestBody or something, so that you could pass None? It might not be a good idea though if you anyway have to specify the type of T, so () could be simpler.